File: 07_covid home/200317_Collection/example#3/java-script/after.js

function forward(){

	console.log('move forward')
}

function backward(){

	console.log('move backward')
}

function left(){

	console.log('move left')
}

function right(){

	console.log('move right')
}


function move(key){

	const dict = {'w':forward, 's':backward, 'a':left, 'd':right}

	const func = dict[key]

	if(func != null)
		func()
}


const key = 's'

move(key)

File: 07_covid home/200317_Collection/example#3/java/after.java
import java.util.HashMap;
import java.lang.reflect.*;



public class Control
{
    
	private HashMap<Character, Method> dict = new HashMap<Character, Method>();
    
    
	public Control(){
        
		try{
			Class cls = Class.forName("Control");
	    
			dict.put('w', cls.getMethod("forward"));
			dict.put('d', cls.getMethod("backward"));
			dict.put('a', cls.getMethod("left"));
			dict.put('d', cls.getMethod("right"));
		}
		catch(Throwable e){
            
		}
	}
    
    
	public void forward() {

		System.out.println("move forward");
	}
	
	public void backward() {

		System.out.println("move backward");
	}
	
	public void left() {

		System.out.println("move left");
	}
	
	public void right() {

		System.out.println("move right");
	}


	public void move(char key) {
	    
		Method method = this.dict.get(key);
	    
		if(method == null)
			return;
	    
		try{
			method.invoke(this);
		}
		catch(Throwable e){
            
		}
	}


	public static void main(String[] args) {
					     
		Control Control = new Control();
		
		Control.move('w');
		Control.move('a');
	}
}
